tests/frontend/project/sourcemirrors/mirror.py

```diff
@@ -0,0 +1,39 @@
+from typing import Optional, Dict, Any
+
+from buildstream import SourceMirror, MappingNode
+
+
+# This mirror plugin basically implements the default behavior
+# by loading the alias definitions as custom "config" configuration
+# instead, and implementing the translate_url method.
+#
+class Sample(SourceMirror):
+    BST_MIN_VERSION = "2.0"
+
+    def configure(self, node):
+        node.validate_keys(SourceMirror.COMMON_CONFIG_KEYS + ["aliases"])
+
+        self.aliases = {}
+
+        aliases = node.get_mapping("aliases")
+        for alias_name, url_list in aliases.items():
+            self.aliases[alias_name] = url_list.as_str_list()
+
+        self.set_supported_aliases(self.aliases.keys())
+
+    def translate_url(
+        self,
+        *,
+        project_name: str,
```

Comment:
The `project_name` parameter has been removed from `SourceMirror.translate_url()`.